sphinx_search/_static/js/rtd_sphinx_search.js

```diff
@@ -71,97 +75,333 @@ const createDomNode = (nodeName, attributes) => {
     return node;
 };
 
+/**
+ * Generate and return html structure
+ * for a page section result.
+ *
+ * @param {Object} sectionData object containing the result data
+ * @param {String} page_link link of the main page. It is used to construct the section link
+ */
+const get_section_html = (sectionData, page_link) => {
+    let section_template =
+        '<a href="<%= section_link %>"> \
+            <div class="outer_div_page_results" id="<%= section_id %>"> \
+                <span class="search__result__subheading"> \
+                    <%= section_subheading %> \
+                </span> \
+                <p class="search__result__content"> \
+                    <%= section_content %> \
+                </p> \
+            </div> \
+        </a> \
+        <br class="br-for-hits">';
+
+    let section_subheading = sectionData._source.title;
+
+    if (
+        sectionData.highlight["sections.title"] !== undefined &&
+        sectionData.highlight["sections.title"].length >= 1
+    ) {
+        section_subheading = sectionData.highlight["sections.title"][0];
+    }
+
+    let section_content =
+        sectionData._source.content.substring(0, 100) + " ...";
```

Comment:
Do we want to substring this here? Perhaps we should do it in the API instead. I don't have a strong feeling here though -- why are we only using 100 chars?

Reply:
I also don't have a particular opinion here.
But I think if we preprocess this on server side, it will reduce the size of the fetched data. So preprocessing this might decrease the time to get the results from server. What are your thoughts on this?

I used the substring with `100` because I think it is just sufficient data from a page which don't have any highlighted content.
We can increase or decrease this as per our choice. I have used this in other places also.